lib/pi_piper/bcm2835.rb

```diff
@@ -1,5 +1,15 @@
-require "pi_piper/bcm2835/version"
+require 'pi_piper/bcm2835/version'
+require 'pi_piper/bcm2835/pin'
+require 'pi_piper/bcm2835/spi'
+require 'pi_piper/bcm2835/i2c'
+require 'pi_piper/bcm2835/driver'
 
 module PiPiper
-  autoload PiPiper::Bcm2835, "pi_piper/bcm2835/bcm2835"
+  module Bcm2835
+    class << self
+      def driver
+        @driver ||= PiPiper::Bcm2835::Driver.new
+      end
```

Comment:
in PiPiper I wrote :
```ruby
  def driver=(klass) 
    if !klass.nil? && (klass <= PiPiper::Driver)
      @driver.close if @driver
      @driver = klass.new
    else
      raise ArgumentError, 'Supply a PiPiper::Driver subclass for driver'
    end
  end
```
It ensure the driver is properly closed before loading a new one. I thought it was enough, and free a bit of memory !